README.md

```diff
@@ -10,17 +10,36 @@ add support for different patterns of non-local control flow to WebAssembly.
 
 * See the [modified spec](https://webassembly.github.io/stack-switching/) for details.
 
-Original `README` from upstream repository follows…
+## Typed Continuations Proposal for WebAssembly
+
+This is a concrete proposal for realising the aims of the stack-switching proposal.
+
+It is based on the [function references](proposals/function-references/Overview.md) and the [tail call](proposals/tail-call/Overview.md) proposals and includes all respective changes.
+
+The proposal is fully [implemented](https://github.com/effect-handlers/wasm-spec) as part of the reference interpreter.
```

Comment:
A reference [implementation](https://github.com/effect-handlers/wasm-spec) is included as an extension to teh reference interpreter.